ci-operator/step-registry/cucushift/installer/rehearse/gcp/ipi/c3-metal/private/provision/cucushift-installer-rehearse-gcp-ipi-c3-metal-private-provision-chain.yaml

```diff
@@ -0,0 +1,22 @@
+chain:
+  as: cucushift-installer-rehearse-gcp-ipi-c3-metal-private-provision
+  steps: 
+  - ref: gcp-provision-c3-metal-network
```

Comment:
Can we ask installer to provision the network resources like a common IPI install after we created the dedicated cluster profile ?

Reply:
As the GCP project doesn't have public DNS zone as the base domain, we have to install private cluster, where a bastion host needs to be set up before the IPI installation.

Reply:
If we have the permission to create network resource, then why not run some similar steps like `cucushift-installer-rehearse-gcp-ipi-private` using the dedicated cluster profile.

Reply:
> Add step gcp-provision-c3-metal-network to read the pre-configured VPC network/subnets settings for subsequent consumers (Note that we use pre-configured VPC here because the existing provisioning VPC step gcp-provision-vpc uses deployment-manager deployments while the GCP project doesn't enable the corresponding service API)

FYI the GCP project doesn't enable the deployment manager API, so the existing step `gcp-provision-vpc` cannot be used.

Reply:
So the network resourced is pre-created and shared to us when we requested the permission for the project, right?

Reply:
No, the network resources were create by me, after getting the permissions in the project, using commands like `gcloud compute networks create...` and etc., rather than `gcloud deployment-manager deployments create...`.

Reply:
FYI before this PR, I do the installations manually, and these network resources were also used.